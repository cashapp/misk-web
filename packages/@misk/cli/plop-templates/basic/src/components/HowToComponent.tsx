import { H1, H3 } from "@blueprintjs/core"
import * as React from "react"
export const HowToComponent = () => {
  return (
    <div>
      <H1>Welcome to your new Misk-Web tab!</H1>
      <p>
        Take a look around! There are some{" "}
        <a href="https://cashapp.github.io/misk-web/docs/guides/building-a-tab/05-pure-vs-connected">
          pure components and connected containers
        </a>
        , examples of form fields, and network requests.
      </p>
      <p>
        For larger examples, check out the{" "}
        <a href="https://github.com/cashapp/misk-web/tree/master/examples/tabs/palette-exemplar">
          palette-exemplar tab in Misk-Web repo
        </a>
        , or play around with the palette-exemplar demo in the{" "}
        <a href="https://cashapp.github.io/misk-web/docs/examples/palette-exemplar">
          docs
        </a>
        .
      </p>
      <H3>
        Good luck building on Misk-Web! See the{" "}
        <a href="https://cashapp.github.io/misk-web/">docs</a> or{" "}
        <a href="https://github.com/cashapp/misk-web">repo</a> to learn more.
      </H3>
    </div>
  )
}

export default HowToComponent
