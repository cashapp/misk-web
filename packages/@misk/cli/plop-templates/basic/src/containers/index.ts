import LoadDataFormContainer from "./LoadDataFormContainer"
import LoadDataTableContainer from "./LoadDataTableContainer"
import TabContainer from "./TabContainer"
export { LoadDataFormContainer, LoadDataTableContainer, TabContainer }
