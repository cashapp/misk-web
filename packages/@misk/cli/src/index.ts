#!/usr/bin/env node

import yargs from "yargs"

yargs
  .scriptName("miskweb")
  .commandDir("commands")
  .demandCommand()
  .option("e", {
    alias: "each",
    describe: "run command in all subdirectories that have miskTab.json",
    type: "boolean"
  })
  .help()
  .version()
  .usage(
    "Misk Web CLI Usage: $ $0 <command> <required arg> [optional arg] [options]"
  )
  .strict().argv
