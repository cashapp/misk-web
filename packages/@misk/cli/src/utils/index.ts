import * as fs from "fs-extra"
import { cd, exec, exit, pwd } from "shelljs"

export * from "./generate"
export * from "./handleCommand"
export * from "./miskTabUtilities"
export * from "./resolveNpmVersion"

export enum MiskPkg {
  "cli" = "@misk/cli",
  "common" = "@misk/common",
  "core" = "@misk/core",
  "dev" = "@misk/dev",
  "prettier" = "@misk/prettier",
  "simpleredux" = "@misk/simpleredux",
  "test" = "@misk/test",
  "tslint" = "@misk/tslint"
}

export interface IMiskTabJSON {
  name: string // name of tab in Title Case
  output_path: string // output path for Webpack build
  port: number // port for Webpack-Dev-Server
  rawGitginore: string // prebuild permanent add to .gitignore file
  rawIndex: boolean // ignore Webpack injection of source links to index.html
  rawPackageJson: any // prebuild permanent add/override to package.json file
  rawTsconfig: any // prebuild permanent add/override to tsconfig.json file
  rawTslint: any // prebuild permanent add/override to tslint.json file
  rawWebpackConfig: any // prebuild permanent add to webpack.config.js file
  relative_path_prefix: string // override default URL for tab: /_tab/{slug}/
  slug: string // unique slug used in URL path
  useWebpackBundleAnalyzer: boolean // turn off/on webpack bundle analyzer reports
  useWebpackExternals: boolean // turn off/on thin build by including externals in Webpack build
  version: string // Misk Web release version or keyword (alpha, latest)
  zipOnBuild: boolean // zip relevant source code of tab into {slug}.tgz after each build
  ___DeprecatedKeys: string // divider to the miskTab.json to improve maintainability
}

export enum Files {
  "gitignore" = ".gitignore",
  "masterDependencies" = "master-dependencies.json",
  "miskTab" = "miskTab.json",
  "old" = ".old_build_files",
  "package" = "package.json",
  "packageLock" = "package-lock.json",
  "prettier" = "prettier.config.js",
  "setupMiskTest" = "setupMiskTest.js",
  "tsconfig" = "tsconfig.json",
  "tslint" = "tslint.json",
  "webpack" = "webpack.config.js",
  "yarnLock" = "yarn.lock"
}

export const JsonOptions = { spaces: 2 }

export const logFormatter = (
  tag: string,
  msg?: string,
  dir: string = pwd().stdout
) =>
  `[${tag.toUpperCase()}][${dir
    .split("/")
    .pop()
    .toUpperCase()}] ${msg}`

export const logDebug = (
  tag: string,
  msg?: string,
  dir: string = pwd().stdout
) => console.log(logFormatter(tag, msg, dir))

export const makePath = (...segments: string[]) => `${segments.join("/")}`

export const parseArgs = (...args: any): { dir: string; rawArgs: any } => ({
  dir: pwd().stdout,
  rawArgs: args
})

export const remove = async (path: string) => {
  try {
    fs.remove(path)
  } catch (e) {
    console.log(`[ERROR] ${e}`)
  }
}

export const execute = (cmd: string, ...args: any) => {
  const { dir } = parseArgs(...args)
  cd(dir)
  const terminal = exec(cmd)
  terminal.stdout
  if (terminal.code) {
    console.log(`\n[ERROR][CODE] Shell command exited with code ${terminal.code}:\n\`${cmd}\``)
    exit(terminal.code)
  }
}

export const npmRunScript = (cmd: string, prebuild: boolean = false) =>
  `${prebuild ? "miskweb prebuild && " : ""}npm run-script ${cmd}`
