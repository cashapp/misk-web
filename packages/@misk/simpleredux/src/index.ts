export * from "./action"
export * from "./deprecated"
import {
  dispatchDefault,
  dispatchSimpleRedux,
  IDispatchOptions,
  IDispatchSimpleRedux,
  ISimpleReduxPayload
} from "./dispatch"
export {
  dispatchDefault,
  dispatchSimpleRedux,
  IDispatchOptions,
  IDispatchSimpleRedux,
  ISimpleReduxPayload
}
export * from "./saga"
export * from "./reducer"
export * from "./utilities"
