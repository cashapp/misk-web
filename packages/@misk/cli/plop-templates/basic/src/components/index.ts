import HowToComponent from "./HowToComponent"
export { HowToComponent }
