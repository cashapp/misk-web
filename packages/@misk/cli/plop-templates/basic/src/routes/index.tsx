import * as React from "react"
import { Route, Switch } from "react-router"
import { TabContainer } from "src/containers"

const routes = (
  <div>
    <Switch>
      <Route path="/_admin/{{name}}/" component={TabContainer} />
      {/* Do not include a Route without a path or it will display during on all tabs: */}
      {/* <Route component={TabContainer} /> */}
    </Switch>
  </div>
)

export default routes
