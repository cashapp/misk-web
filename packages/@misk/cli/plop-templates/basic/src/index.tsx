import { createApp, createIndex } from "@misk/core"
import * as Ducks from "./ducks"
import routes from "./routes"
export * from "./components"
export * from "./containers"

createIndex("{{name}}", createApp(routes), Ducks)
