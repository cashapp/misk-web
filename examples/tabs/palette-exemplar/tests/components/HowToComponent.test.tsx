import React from "react"
import { cleanup, render } from "@testing-library/react"
import { HowToComponent } from "src/components"

describe("HowToComponent", () => {
  afterEach(cleanup)
  it("HowToComponent can render", () => {
    const { asFragment } = render(<HowToComponent />)
    expect(asFragment()).toMatchSnapshot()
  })
})
