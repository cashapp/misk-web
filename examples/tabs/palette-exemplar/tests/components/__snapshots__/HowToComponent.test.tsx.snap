// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToComponent HowToComponent can render 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="bp4-heading"
    >
      Redux Powered Containers
    </h1>
    <p>
      There are examples below of forms, networks, buttons, tables, and other UI you will find helpful in building your Misk-Web tab. If you're looking for a way to test network functionality, HttpBin.org has a helpful echo Docker container that will do the trick.
    </p>
    <ol>
      <li>
        <code>
          $ docker run -p 1080:80 kennethreitz/httpbin
        </code>
      </li>
      <li>
        Use 
        <code>
          http://localhost:1080/anything
        </code>
         as your URL for any URL requests below
      </li>
    </ol>
    <h3
      class="bp4-heading"
    >
      Good luck building on Misk-Web! See the 
      <a
        href="https://cashapp.github.io/misk-web/"
      >
        docs
      </a>
       or 
      <a
        href="https://github.com/cashapp/misk-web"
      >
        repo
      </a>
       to learn more.
    </h3>
  </div>
</DocumentFragment>
`;
